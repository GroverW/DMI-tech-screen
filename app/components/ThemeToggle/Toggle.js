/**
 * A dropdown selector
 */

import styled from 'styled-components';

const Toggle = styled.select`
  background-color: ${props => props.theme.background};
  color: ${props => props.theme.font};
  border: 1px solid ${props => props.theme.buttonBorder};

  &:hover {
    cursor: pointer;
  }
`;

export default Toggle;
