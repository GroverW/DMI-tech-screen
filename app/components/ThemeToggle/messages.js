/*
 * ThemeToggle Messages
 *
 * This contains all the text for the ThemeToggle component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.ThemeToggle';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the ThemeToggle component!',
  },
});
