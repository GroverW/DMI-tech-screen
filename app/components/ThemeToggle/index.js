/**
 *
 * ThemeToggle
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

// import { FormattedMessage } from 'react-intl';
import { makeGetTheme } from '../../containers/Themes/selectors';
import { themes } from '../../containers/Themes';
import { setTheme } from '../../containers/Themes/actions';
import Toggle from './Toggle';

function ThemeToggle(props) {
  return (
    <div>
      <Toggle
        theme={props.theme.colors}
        value={props.theme.name}
        onChange={props.onThemeChange}
      >
        {Object.keys(themes).map(theme => (
          <option key={theme} id={theme} value={theme}>
            {theme}
          </option>
        ))}
      </Toggle>
    </div>
  );
}

ThemeToggle.propTypes = {
  onThemeChange: PropTypes.func,
  theme: PropTypes.object,
};

const mapStateToProps = createSelector(
  makeGetTheme(),
  theme => ({
    theme,
  }),
);

export function mapDispatchToProps(dispatch) {
  return {
    onThemeChange: evt => dispatch(setTheme(evt.target.value)),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ThemeToggle);
