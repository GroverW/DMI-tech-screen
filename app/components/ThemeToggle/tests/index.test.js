/**
 *
 * Tests for ThemeToggle
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../configureStore';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import ThemeToggle from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<ThemeToggle />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ThemeToggle />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ThemeToggle />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
