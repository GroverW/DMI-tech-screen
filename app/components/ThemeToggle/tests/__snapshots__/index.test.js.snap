// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeToggle /> Should render and match the snapshot 1`] = `
<div>
  <select
    class="Toggle-oqiqhq-0 jVmmhU"
  >
    <option
      id="light"
      value="light"
    >
      light
    </option>
    <option
      id="dark"
      value="dark"
    >
      dark
    </option>
  </select>
</div>
`;
